### PR TITLE
Add code check for framework.ExpectEqual()

### LINKED ISSUE #79686: Use framework.ExpectEqual() for e2e tests
**What happened**:

After fixing golint failures under `test/e2e`, we need to put `gomega` for each `Expect()` and `Equal()`.
Then common error checking code has become like:

`gomega.Expect(foo).To(gomega.Equal(bar))`

We can replace this code with `framework.ExpectEqual(foo, bar)` which is more readable and easy to be understood what the code does.

Now we have following e2e test directories which can be replaced:
- [x] test/e2e/apimachinery (https://github.com/kubernetes/kubernetes/pull/79685)
- [x] test/e2e/apps (https://github.com/kubernetes/kubernetes/pull/79695)
- [x] test/e2e/auth (https://github.com/kubernetes/kubernetes/pull/79762)
- [x] test/e2e/autoscaling (https://github.com/kubernetes/kubernetes/pull/79765)
- [x] test/e2e/cloud (https://github.com/kubernetes/kubernetes/pull/79821)
- [x] test/e2e/common (https://github.com/kubernetes/kubernetes/pull/79769)
- [ ] test/e2e/framework (https://github.com/kubernetes/kubernetes/pull/79771)
- [x] test/e2e/kubectl (https://github.com/kubernetes/kubernetes/pull/79699)
- [x] test/e2e/lifecycle (https://github.com/kubernetes/kubernetes/pull/79860)
- [x] test/e2e/network (https://github.com/kubernetes/kubernetes/pull/79791)
- [x] test/e2e/node (https://github.com/kubernetes/kubernetes/pull/79792)
- [x] test/e2e/scalability (https://github.com/kubernetes/kubernetes/pull/79842)
- [x] test/e2e/scheduling (https://github.com/kubernetes/kubernetes/pull/79781)
- [x] test/e2e/servicecatalog (https://github.com/kubernetes/kubernetes/pull/79754)
- [x] test/e2e/storage (https://github.com/kubernetes/kubernetes/pull/79755)
- [x] test/e2e/upgrades (https://github.com/kubernetes/kubernetes/pull/79756)
- [x] test/e2e/windows (https://github.com/kubernetes/kubernetes/pull/79756)
- [x] test/e2e_node/apparmor_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/container_log_rotation_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/container_manager_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/critical_pod_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/device_plugin_test.go (https://github.com/kubernetes/kubernetes/pull/80790)
- [x] test/e2e_node/eviction_test.go (https://github.com/kubernetes/kubernetes/pull/80843)
- [x] test/e2e_node/gpu_device_plugin_test.go (https://github.com/kubernetes/kubernetes/pull/80843)
- [x] test/e2e_node/mirror_pod_test.go (https://github.com/kubernetes/kubernetes/pull/80843)
- [x] test/e2e_node/util.go (https://github.com/kubernetes/kubernetes/pull/80843)
- [x] test/e2e_kubeadm/dns_addon_test.go (https://github.com/kubernetes/kubernetes/pull/80795)
- [x] test/e2e_kubeadm/kubeadm_certs_test.go (https://github.com/kubernetes/kubernetes/pull/80795)
- [x] test/e2e_kubeadm/proxy_addon_test.go (https://github.com/kubernetes/kubernetes/pull/80795)

### PR DESCRIPTION
**Pull request by oomichi**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This adds code check for using framework.ExpectEqual(foo, bar) in
e2e tests for keeping the test code simple.

Close: https://github.com/kubernetes/kubernetes/issues/79686

**Special notes for your reviewer**:

When we finish all tasks in https://github.com/kubernetes/kubernetes/issues/79686 this PR can be merged.

### CODE REVIEW

File: hack/verify-test-code.sh [hunk 1]
```diff
@@ -42,6 +42,15 @@ do
     fi
 done
 
+errors_expect_equal=()
+for file in "${all_e2e_files[@]}"
+do
+    if grep "Expect(.*)\.To(.*Equal(" "${file}" > /dev/null
```
Comment (k-toyoda-pi) [inline hack/verify-test-code.sh]: The line detects test/e2e/auth/node_authn.go as containing errors.
node_authn.go contains the following line. But I think that it should not be detected.
```
gomega.Expect(result).To(gomega.Or(gomega.Equal("401"), gomega.Equal("403")), ...
```
How about changing it like `grep -E "Expect\(.*\)\.To\((gomega\.Equal|Equal)` ?

Comment (oomichi) [reply hack/verify-test-code.sh]: @k-toyoda-pi Thank you for your comment, nice catch.
I will update this soon.

**Comment by spiffxp** [comment]
/priority important-soon
/retest
Can we find a way to do these checks without bash?

**Comment by oomichi** [comment]
@spiffxp 
> Can we find a way to do these checks without bash?

Humm, TBH I cannot imagine alternative way except current bash way :-(

Anyways, this cannot be merged at this time because 2 PRs(https://github.com/kubernetes/kubernetes/pull/80790 and not-yet PR according to https://github.com/kubernetes/kubernetes/issues/79686) are necessary to be merged first before this.

**Comment by oomichi** [comment]
To wait for merging necessary PRs before this.

/hold

**Comment by oomichi** [comment]
All necessary PRs are merged, so let's retest this.

/retest
/hold cancel

**Comment by oomichi** [comment]
@spiffxp This seems ready at this time, PTAL :-)

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Pull request merged**